samples/isolation-forest-clustering/Training.ipynb

```diff
@@ -0,0 +1,287 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from platiagro import load_dataset\n",
+    "\n",
+    "df = load_dataset(name=dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load metadata about the dataset\n",
+    "For example, below we get the feature type for each column in the dataset. (eg. categorical, numerical, or datetime)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from platiagro import stat_dataset\n",
+    "\n",
+    "metadata = stat_dataset(name=dataset)\n",
+    "featuretypes = metadata[\"featuretypes\"]\n",
+    "featuretypes = np.array(featuretypes)\n",
+    "columns = df.columns.to_numpy()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Remove datetime features\n",
+    "Datetime columns require separate preprocessing or feature extraction steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from platiagro.featuretypes import DATETIME\n",
+    "\n",
+    "datetime_indexes = (featuretypes == DATETIME)\n",
+    "\n",
+    "if sum(datetime_indexes) > 0:\n",
+    "    df = df[:, np.where(~datetime_indexes)[0]]\n",
+    "    featuretypes = np.delete(featuretypes, np.where(datetime_indexes))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fit a model using sklearn.ensemble.IsolationForest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from platiagro.featuretypes import NUMERICAL\n",
+    "from sklearn.compose import ColumnTransformer\n",
+    "from sklearn.impute import SimpleImputer\n",
+    "from sklearn.ensemble import IsolationForest\n",
+    "from sklearn.pipeline import Pipeline\n",
+    "from sklearn.preprocessing import OrdinalEncoder\n",
+    "\n",
+    "numerical_indexes = (featuretypes == NUMERICAL)\n",
+    "\n",
+    "pipeline = Pipeline(steps=[\n",
+    "            ('handle_missing_values',\n",
+    "             ColumnTransformer(\n",
+    "                [('imputer_mean', SimpleImputer(strategy='mean'), numerical_indexes),\n",
+    "                 ('imputer_mode', SimpleImputer(strategy='most_frequent'), ~numerical_indexes)],\n",
+    "                 remainder='passthrough')),\n",
+    "            ('handle_categorical_features',\n",
+    "                 ColumnTransformer(\n",
+    "                         [('feature_encoder', OrdinalEncoder(), ~numerical_indexes)])),\n",
+    "            ('estimator', IsolationForest(max_samples=max_samples,\n",
+    "                            contamination=contamination,\n",
+    "                            max_features=max_features))\n",
```

Comment:
@lucaslzl 
Existe um bug no `handle_categorical_features` que o Samuel descobriu outro dia:
O passo anterior, `handle_missing_values` reordena as features e coloca as numéricas no início e as não numéricas no fim. 
Então o passo `('feature_encoder', OrdinalEncoder(), ~numerical_indexes)` vai usar um `numerical_indexes` que pode não ser correto.

O Samuel propôs uma correção aqui:
https://github.com/platiagro/projects/blob/master/samples/svr/Training.ipynb

**Que cria um "non_numerical_indexes_after_handle_missing_values".**
Use-o como exemplo.

**Ele também recomendou usar o category_encoder do sklearn-contrib po [motivos descritos aqui](https://github.com/platiagro/projects/pull/59#issuecomment-633352415).

`from category_encoders.ordinal import OrdinalEncoder`